Sources_v3/Workers/MessageUpdater.swift

```diff
@@ -197,6 +197,68 @@ class MessageUpdater<ExtraData: ExtraDataTypes>: Worker {
             }
         }
     }
+    
+    /// Flags or unflags the message with the provided `messageId` depending on `flag` value.
+    /// If the message doesn't exist locally it will be fetched and saved locally first first.
+    ///
+    /// - Parameters:
+    ///   - flag: The indicator saying whether the messageId should be flagged or unflagged.
+    ///   - messageId: The identifier of a messageId that should be flagged or unflagged.
+    ///   - completion: Called when the API call is finished. Called with `Error` if the remote update fails.
+    ///
+    func flagMessage(_ flag: Bool, with messageId: MessageId, in cid: ChannelId, completion: ((Error?) -> Void)? = nil) {
+        fetchAndSaveMessageIfNeeded(messageId, cid: cid) { error in
```

Comment:
I have no idea why we do this here 🤔

Reply:
It's here because it might happen the message does not exist locally -> if there is no message in the database, there is nothing to **add/remove** from `currentUser.flaggedMessages`. 

We have to make sure the message is loaded before we flag/unflag it. I thought it's the right place, because end users can invoke `controller.flag/unflag()` without invoking `controller.synchronize`.

We have nearly the same thing in `ChannelMemberListUpdater` where we fetch and save a channel if needed before queuing the members.

Reply:
-> Slack